### PR TITLE
Remove DeepGEMM for indexer GEMM in piecewise NSA path

### PR DESCRIPTION
**Pull request by b8zhong**
## Motivation
Remove it in piecewise CUDA graph path, which seems to be accidentally reintroduced in https://github.com/sgl-project/sglang/pull/23351 due to PRs merging at different times. Additionally, it will increase the warmup time

## Modifications
Remove it.

## Accuracy Tests
It's already been validated in https://github.com/sgl-project/sglang/pull/23856

## Speed Tests and Profiling
Unit benchmark with CUPTI. 
Results: (only N = 64 and N = 32 are relevant for indexer for DSV3.2 and GLM-5), the rest are Deepseek V4 shapes, which shows we could remove it from Deepseek V4 path later (currently not used on default path anyway)
image

---
### CI States

Latest PR Test (Base): :no_entry_sign: [Run #26527866613](https://github.com/sgl-project/sglang/actions/runs/26527866613)
Latest PR Test (Extra): :x: [Run #26527866300](https://github.com/sgl-project/sglang/actions/runs/26527866300)

### CODE REVIEW

File: python/sglang/srt/layers/attention/dsa/dsa_indexer.py [hunk 1]
```diff
@@ -162,12 +162,7 @@ def logits_head_gate_pcg(
         softmax_scale: float,
         q_scale: torch.Tensor,
     ) -> torch.Tensor:
-        from sglang.srt.layers.deep_gemm_wrapper import entrypoint as deep_gemm_wrapper
-
-        out = torch.empty(
-            (x.shape[0], weight.shape[0]), dtype=torch.float32, device=x.device
-        )
-        deep_gemm_wrapper.gemm_nt_bf16bf16f32(x, weight, out)
+        out = torch.mm(x, weight.t(), out_dtype=torch.float32)
```
[the remark this answers is not in the document]
Comment (b8zhong) [reply python/sglang/srt/layers/attention/dsa/dsa_indexer.py:165]: That's not true. It's supported in 2.10

**Review by Fridge003 — APPROVED** [review]
